### DISCUSSION TITLE
Updating git identity on terminal.

### DISCUSSION THREAD

**Discussion by NdutaR3Corda** [community/community#40714] [New to GitHub]
Hello. I need help for the sake of my coursework. I am trying to do a git configuration and this is the error I get. How do I resolve this?

root@e368949080e9:~/alx-zero_day# git config --global user.email "ndutaihrene16@gmail.com"                                      
root@e368949080e9:~/alx-zero_day# git config --global user.name "Nduta Mwangi"                                                  
warning: user.name has multiple values                                                                                          
error: cannot overwrite multiple values with a single value                                                                     
       Use a regexp, --add or --replace-all to change user.name.                                                                
root@e368949080e9:~/alx-zero_day# git config --global user.name "NdutaR3Corda"                                                  
warning: user.name has multiple values                                                                                          
error: cannot overwrite multiple values with a single value                                                                     
       Use a regexp, --add or --replace-all to change user.name.                                                                
root@e368949080e9:~/alx-zero_day#

**Comment by airtower-luna** [comment]
The message indicates that your Git config file (usually `~/.gitconfig`) is broken, and has multiple entries for `user.name`. This has nothing to do with multiple words in your name, that's perfectly fine.

Either use the `--replace-all` option as the message indicates to remove all the existing entries and replace them with the one value you're setting, or check and fix the file manually.

That said, the command line indicates you're working as root (AKA the "super user"). You generally should *not* do that, because it's bad for security. Only commands that really require it should run as root, and Git doesn't. The hostname looks like you might be working in a container, which somewhat limits the damage, but you should still avoid it.

**Reply by NdutaR3Corda** [reply]
Two questions.
1. Being a noob, please guide me on how to correctly input the _-`-replace-all_` command. I have a feeling am doing it wrong.
2. How to change from a superuser? Am using the intranet and I don't know how to go about it. The su command isnt changing anything.

**Reply by airtower-luna** [reply]
> please guide me on how to correctly input the _--replace-all_ command.

It's not a separate *command*, it's an *option* you add to the `git config` command. For example:

```
git config --global --replace-all user.name "Nduta Mwangi"
```

> How to change from a superuser? Am using the intranet and I don't know how to go about it. The su command isnt changing anything.

Usually you need to actively *do* something to become the superuser (like logging in). What to change really depends on your setup.

**Reply by NdutaR3Corda** [reply]
Thank you so much for your help.

**Reply by airtower-luna** [reply]
Glad I could help! Please don't forget to mark the answer. :slightly_smiling_face:

**Comment by mohamedzhioua** [comment]
to set your account's default identity globally run below commands

git config --global user.email "you@example.com"
git config --global user.name "Your Name"
git config --global user.password "your password"
To set the identity only in current repository , remove --global and run below commands in your Project/Repo root directory

git config user.email "you@example.com"
git config user.name "Your Name"
git config user.password "your password"
Example:

email -> organization email Id
name  -> mostly <employee Id> or <FirstName, LastName>

**Reply by NdutaR3Corda** [reply]
with or without quotes?
my git username at user.name?
my PAT at user.password?

**Reply by mohamedzhioua** [reply]
visit the linlk below :
https://www.theserverside.com/blog/Coffee-Talk-Java-News-Stories-and-Opinions/How-set-git-config-username-and-email-fields-in-global-and-local-configuration-files

**Reply by airtower-luna** [reply]
There is no `user.password` option, see the [`git config` documentation](https://git-scm.com/docs/git-config). If you set it anyway you're needlessly putting your password on disk in plain text, which is a security risk.

`user.email` and `user.name` are *exclusively* for the author/committer data that will be recorded in your commits. They have only one thing to do with your GitHub account: If the email address (`user.email`) is connected to your account, GitHub will recognize it as yours.

**Comment by mohamedzhioua** [comment]
without Quotes yeah and yes user.name / user.password @NdutaR3Corda
